dv-export/python/data_store.py

```diff
@@ -0,0 +1,139 @@
+import logging
+from abc import ABC, abstractmethod
+import pandas as pd
+
+from sqlalchemy import (Column, Table, create_engine, MetaData, Integer,
+                        String, Numeric, BigInteger, Boolean, insert, text)
+from dv_export_model import DVEXportDataType, ColumnInfo, TableInfo
+
+logger = logging.getLogger("data_store")
+
+
+def convert_data_types_to_df_dtype_dict(column_infos) -> dict:
+    column_name_to_dtype = {}
+    for column in column_infos:
+        dtype = ''
+        if column.data_type == DVEXportDataType.Date or column.data_type == DVEXportDataType.Integer:
+            dtype = 'Int64'
+        elif column.data_type == DVEXportDataType.String:
+            dtype = 'object'
+        elif column.data_type == DVEXportDataType.Number:
+            dtype = 'float64'
+        elif column.data_type == DVEXportDataType.Boolean:
+            dtype = 'boolean'
+        column_name_to_dtype[column.name] = dtype
+    return column_name_to_dtype
+
+
+class DataStore(ABC):
+
+    @abstractmethod
+    def create_initial_table_definition(self, tbl: TableInfo):
+        pass
+
+    @abstractmethod
+    def drop_table(self, tbl_name: str):
+        pass
+
+    @abstractmethod
+    def get_data_store_data_type_from_dv_export_data_type(self, data_type: DVEXportDataType):
+        pass
+
+    @abstractmethod
+    def insert_from_files_into_table(self,
+                                     tbl_name: str,
+                                     column_infos: list[ColumnInfo],
+                                     file_paths_for_tbl: list[str]):
+        pass
+
+    @abstractmethod
+    def drop_column(self, tbl_name: str, column_name: str):
+        pass
+
+
+class SQLAlchemyDataStore(DataStore):
+    def __init__(self, url, echo=False):
+        self.engine = create_engine(url, echo=echo)
+        self.metadata = MetaData()
+        self.metadata.reflect(self.engine)
+
+    def get_data_store_data_type_from_dv_export_data_type(self, data_type: DVEXportDataType):
+        if data_type == DVEXportDataType.String:
+            return String()
+        elif data_type == DVEXportDataType.Integer:
+            return Integer()
+        elif data_type == DVEXportDataType.Number:
+            return Numeric()
+        elif data_type == DVEXportDataType.Date:
+            return BigInteger()
+        elif data_type == DVEXportDataType.Boolean:
+            return Boolean()
+        else:
+            raise NotImplementedError(f"data_type={data_type} has no corresponding data type for SQLAlchemy")
+
+    def get_list_of_sql_columns(self, column_infos: list[ColumnInfo]) -> list[Column]:
+        columns = []
+        for column_info in column_infos:
+            name = column_info.name
+            sql_data_type = self.get_data_store_data_type_from_dv_export_data_type(column_info.data_type)
+
+            # todo: Hack until https://visiercorp.atlassian.net/browse/VAN-120816 is fixed
+            if column_info.name == 'Location_LEVEL':
```

Comment:
Until this is fixed I think should just have nullable=true instead of linking internal Jira here